express-notes-api/index.js

```diff
@@ -0,0 +1,85 @@
+var express = require('express');
+var app = express();
+var jsonExpress = express.json();
+app.use(jsonExpress);
+
+app.get('/api/notes', function (req, res) {
+  var dataFetch = require('./data.json');
+  const noteArray = [];
+  for (const i in dataFetch.notes) {
+    noteArray.push(dataFetch.notes[i]);
+  }
+  res.json(noteArray);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  var dataFetch = require('./data.json');
+  if (req.params.id < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (dataFetch.notes[req.params.id] === undefined) {
+    res.status(404).json({ error: 'cannot find note with id ' + req.params.id });
+  } else {
+    res.json(dataFetch.notes[req.params.id]);
+  }
+});
+
+app.post('/api/notes', function (req, res) {
+  var reqBody = req.body;
+  if (reqBody.content === undefined) {
+    res.status(400).json({ error: 'content is a required field' });
+  } else {
+    var dataForPost = require('./data.json');
```

Comment:
You should be able to `require()` your `data.json` file one time at the top of `index.js`.